src/main/kotlin/protein/kotlinbuilders/KotlinApiBuilder.kt

```diff
@@ -249,79 +255,84 @@ class KotlinApiBuilder(
+      else -> getKotlinClassTypeName(property.type, property.format).requiredOrNullable(property.required)
     }
   }
 
   private fun getMethodParameters(
     operation: MutableMap.MutableEntry<HttpMethod, Operation>
   ): Iterable<ParameterSpec> {
-    val methodParameters = ArrayList<ParameterSpec>()
-    for (parameter in operation.value.parameters) {
+    return operation.value.parameters.mapNotNull { parameter ->
+      // Transform parameters in the format foo.bar to fooBar
+      val name = parameter.name.split('.').mapIndexed { index, s -> if (index > 0) s.capitalize() else s }.joinToString("")
       when (parameter.`in`) {
         "body" -> {
-          val bodyParameterSpec: ParameterSpec = getBodyParameterSpec(parameter)
-          methodParameters.add(bodyParameterSpec)
+          ParameterSpec.builder(name, getBodyParameterSpec(parameter))
+            .addAnnotation(AnnotationSpec.builder(Body::class).build()).build()
         }
         "path" -> {
-          val pathParameterSpec =
-            ParameterSpec.builder(parameter.name, getKotlinClassTypeName((parameter as PathParameter).type))
-              .addAnnotation(
-                AnnotationSpec.builder(Path::class).addMember("\"${parameter.name}\"").build()).build()
-          methodParameters.add(pathParameterSpec)
+          val type = getKotlinClassTypeName((parameter as PathParameter).type, parameter.format).requiredOrNullable(parameter.required)
+          ParameterSpec.builder(name, type)
+            .addAnnotation(AnnotationSpec.builder(Path::class).addMember("\"${parameter.name}\"").build()).build()
         }
         "query" -> {
-          val queryParameterSpec: ParameterSpec = if ((parameter as QueryParameter).type == ARRAY_SWAGGER_TYPE) {
-            ParameterSpec.builder(
-              parameter.name, List::class.asClassName().parameterizedBy(getKotlinClassTypeName(parameter.items.type))
-            ).addAnnotation(
-                AnnotationSpec.builder(Query::class).addMember("\"${parameter.name}\"").build()).build()
+          if ((parameter as QueryParameter).type == ARRAY_SWAGGER_TYPE) {
+            val type = List::class.asClassName().parameterizedBy(getKotlinClassTypeName(parameter.items.type)).requiredOrNullable(parameter.required)
+            ParameterSpec.builder(name, type)
           } else {
-            ParameterSpec.builder(parameter.name, getKotlinClassTypeName(parameter.type))
-              .addAnnotation(
-                AnnotationSpec.builder(Query::class).addMember("\"${parameter.name}\"").build()).build()
-          }
-          methodParameters.add(queryParameterSpec)
+            val type = getKotlinClassTypeName(parameter.type, parameter.format).requiredOrNullable(parameter.required)
+            ParameterSpec.builder(name, type)
+          }.addAnnotation(AnnotationSpec.builder(Query::class).addMember("\"${parameter.name}\"").build()).build()
         }
+        else -> null
```

Comment:
Returning "null" may lead to issues, also you're not making the returned value as nullable. I think it would be better to return an empty list to avoid possible crashes.

Reply:
The return is the map of the parameters (line #274). If the list is empty, it will return an empty list.

Reply:
Seems that I did not read it well 😅